src/utils/kubeclient.ts

```diff
@@ -53,3 +55,97 @@ export function createKubeClient(config: string | AppConfig, context?: string) {
 
   return kc;
 }
+
+function parseCanI(stdout: string, namespace: string): ClusterAccess {
+  const lines = stdout.split('\n');
+
+  const permissions: KubePermissions[] = [];
+  let hasFullAccess = false;
+
+  lines.forEach((line, index) => {
+    if (!index) {
+      return;
+    }
+    const columns = line.split(/\s{2,100}/);
+
+    const [resourceName, , , rawVerbs] = columns;
+
+    if (!resourceName) {
+      return;
+    }
+
+    const cleanVerbs = (rawVerbs as string)
+      .replace('[', '')
+      .replace(']', '');
+
+    if (resourceName === '*.*' && cleanVerbs === '*') {
+      hasFullAccess = true;
+    }
+
+    const verbs = cleanVerbs ? cleanVerbs.split(' ') : [];
+
+    permissions.push({
+      resourceName,
+      verbs,
+    });
+  });
+
+  return {
+    permissions,
+    hasFullAccess,
+    namespace,
+  };
+}
+
+export function getKubeAccess(namespace: string): ClusterAccess {
+  // execSync(`kubectl config use-context ${currentContext}`);
+  const command = `kubectl auth can-i --list --namespace=${namespace}`;
+  const canStdOut = execSync(command).toString();
+  return parseCanI(canStdOut, namespace);
+}
+
+export function hasAccessToResource(resourceName: string, verb: string, clusterAccess?: ClusterAccess) {
```

Comment:
same question as with `KubePermissions`, is the `resourceName` enough?

Reply:
yes, resource name and verb